Sources/SwiftyRequest/RestRequest.swift

```diff
@@ -797,14 +805,37 @@ extension RestRequest: URLSessionDelegate {
         let method = challenge.protectionSpace.authenticationMethod
         let host = challenge.protectionSpace.host
 
-        guard let url = URLComponents(string: self.url), let baseHost = url.host else {
+        guard let url = URLComponents(string: self.url), let baseHost = url.host, let certificatePath = self.certificatePath else {
```

Comment:
This doesn't look right, we'll never get to the switch if `certificatePath` is nil.